server/src/main/java/org/opensearch/action/support/HandledTransportAction.java

```diff
@@ -75,19 +76,59 @@ protected HandledTransportAction(
         ActionFilters actionFilters,
         Writeable.Reader<Request> requestReader
     ) {
-        this(actionName, canTripCircuitBreaker, transportService, actionFilters, requestReader, ThreadPool.Names.SAME);
+        this(actionName, canTripCircuitBreaker, null, transportService, actionFilters, requestReader, ThreadPool.Names.SAME);
     }
 
     protected HandledTransportAction(
         String actionName,
         boolean canTripCircuitBreaker,
+        AdmissionControlActionType admissionControlActionType,
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Writeable.Reader<Request> requestReader
+    ) {
+        this(
+            actionName,
+            canTripCircuitBreaker,
+            admissionControlActionType,
+            transportService,
+            actionFilters,
+            requestReader,
+            ThreadPool.Names.SAME
+        );
+    }
+
+    protected HandledTransportAction(
+        String actionName,
+        boolean canTripCircuitBreaker,
+        AdmissionControlActionType admissionControlActionType,
         TransportService transportService,
         ActionFilters actionFilters,
         Writeable.Reader<Request> requestReader,
         String executor
     ) {
         super(actionName, actionFilters, transportService.getTaskManager());
-        transportService.registerRequestHandler(actionName, executor, false, canTripCircuitBreaker, requestReader, new TransportHandler());
+
+        if (admissionControlActionType != null) {
```

Comment:
Can we remove if else and use constructor override to register handlers ? Below overrides will cover your usecase.
```
    protected HandledTransportAction(
        String actionName,
        boolean canTripCircuitBreaker,
        AdmissionControlActionType admissionControlActionType,
        TransportService transportService,
        ActionFilters actionFilters,
        Writeable.Reader<Request> requestReader
    ) {
        this(actionName, canTripCircuitBreaker, admissionControlActionType, transportService, actionFilters, requestReader, ThreadPool.Names.SAME);
    }

    protected HandledTransportAction(
        String actionName,
        boolean canTripCircuitBreaker,
        AdmissionControlActionType admissionControlActionType,
        TransportService transportService,
        ActionFilters actionFilters,
        Writeable.Reader<Request> requestReader,
        String executor
    ) {
        super(actionName, actionFilters, transportService.getTaskManager());
        transportService.registerRequestHandler(actionName, executor, false, canTripCircuitBreaker, admissionControlActionType, requestReader, new TransportHandler());
    }
```

Reply:
Removed the null check from constructor, modified `transportService.registerRequestHandler` to create handlers based on null value